code/src/Components/MovieDetails.js

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react'
+import { useParams, Link } from 'react-router-dom'
+import './movie-details.css'
+
+export const MovieDetails = () => {
+  const { movieId } = useParams()
+
+  const [details, setDetails] = useState(null)
```

Comment:
empty object would be more appropriate as a default state :

```
const [details, setDetails] = useState({})
```

Later on, when we check if we need to return `null` instead of JSX, we can check it like this :

```
const isObjectEmpty = (obj) => {
  for(var prop in obj) {
    if(obj.hasOwnProperty(prop)) {
      return false;
    }
  }

  return JSON.stringify(obj) === JSON.stringify({});
}

if (isObjectEmpty(details)) {
  return null;
}
```

We have to do it like this, because by default empty object is converted by JavaScript to `true`